applications/MeshingApplication/custom_utilities/tet10_refinement_utility.h

```diff
@@ -0,0 +1,499 @@
+    */
+    void LocalRefineTet10Mesh(bool refine_on_reference, bool interpolate_internal_variables) {
+            for (auto element : mModelPart.Elements()) element.SetValue(SPLIT_ELEMENT,true);
+            for (auto condition : mModelPart.Conditions()) condition.SetValue(SPLIT_ELEMENT,true);
+            LocalRefineMesh(refine_on_reference, interpolate_internal_variables);
+        } 
+  
+    
+protected:
+    ///@name Protected static Member Variables
+    ///@{
+    //int mPreviousRefinementLevel;
+    //int mCurrentRefinementLevel;
+    ///@}
+    ///@name Protected member Variables
+    ///@{
+
+    ///@}
+    ///@name Protected Operators
+    ///@{
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+    ///@}
+    ///@name Protected  Access
+    ///@{
+
+    ///@}
+    ///@name Protected Inquiry
+    ///@{
+
+    ///@}
+    ///@name Protected LifeCycle
+    ///@{
+    ///@}
+
+private:
+    ///@name Private static Member Variables
+    ///@{
+
+    ///@}
+    ///@name Private member Variables
+    ///@{
+
+    ///@}
+    ///@name Private Operators
+    ///@{
+
+    ///@}
+    ///@name Private Operations
+    ///@{
+        /**
+    * It erases the old elements and it creates the new ones
+    * @param Coord: The compressed matrix containing at (i,j) the id of the node created between nodes i,j
+    * @param New_Elements: The new elements created
+    * @param interpolate_internal_variables: A boolean that defines if it is necessary to interpolate the internal variables
+    * @return this_model_part: The model part of the model (it is the input too)
+    */
+
+    void EraseOldElementAndCreateNewElement(
+            ModelPart& this_model_part,
+            const compressed_matrix<int>& Coord,
+            PointerVector< Element >& NewElements,
+            bool interpolate_internal_variables
+    ) override
+    {
+	ElementsArrayType& rElements = this_model_part.Elements();
+	ElementsArrayType::iterator it_begin = rElements.ptr_begin();
+	ElementsArrayType::iterator it_end = rElements.ptr_end();
+	unsigned int to_be_deleted = 0;
+	unsigned int large_id = (rElements.end() - 1)->Id() * 15;
+	unsigned int current_id = (rElements.end() - 1)->Id() + 1;
+	bool create_element = false;
+	int number_elem = 0;
+	int splitted_edges = 0;
+	int internal_node = 0;
+
+	const ProcessInfo& rCurrentProcessInfo = this_model_part.GetProcessInfo();
+	int edge_ids[6];
+	int t[56];
+	std::vector<int> aux;
+
+        std::cout << "****************** REFINING MESH ******************" << std::endl;
+        std::cout << "OLD NUMBER ELEMENTS: " << rElements.size() << std::endl;
+
+	for (int & i : t)
+	{
+	    i = -1;
+	}
+
+	for (ElementsArrayType::iterator it = it_begin; it != it_end; ++it)
+	{
+	  Element::GeometryType& geom = it->GetGeometry();
+          CalculateEdges(geom, Coord, edge_ids, aux);
+
+	  create_element = TetrahedraSplit::Split_Tetrahedra(edge_ids, t, &number_elem, &splitted_edges, &internal_node);
+
+	  if (create_element == true)
```

Comment:
In your case should be always true. In fact I would send an error if all the edges for this element don't have their corresponding nodes created